pimd/pim_iface.c

```diff
@@ -1593,24 +1601,21 @@ int pim_ifp_up(struct interface *ifp)
 	}
 
 	pim = pim_get_pim_instance(ifp->vrf_id);
-	if (if_is_operative(ifp)) {
-		struct pim_interface *pim_ifp;
 
-		pim_ifp = ifp->info;
-		/*
-		 * If we have a pim_ifp already and this is an if_add
-		 * that means that we probably have a vrf move event
-		 * If that is the case, set the proper vrfness.
-		 */
-		if (pim_ifp)
-			pim_ifp->pim = pim;
+	pim_ifp = ifp->info;
+	/*
+	 * If we have a pim_ifp already and this is an if_add
+	 * that means that we probably have a vrf move event
+	 * If that is the case, set the proper vrfness.
+	 */
+	if (pim_ifp)
+		pim_ifp->pim = pim;
 
-		/*
-		  pim_if_addr_add_all() suffices for bringing up both IGMP and
-		  PIM
-		*/
-		pim_if_addr_add_all(ifp);
-	}
+	/*
+	  pim_if_addr_add_all() suffices for bringing up both IGMP and
+	  PIM
+	*/
+	pim_if_addr_add_all(ifp);
```

Comment:
should we move this also inside if(pim_ifp) condition which is above this line? That way need not enter into pim_if_addr_add_all to figure out we have pim_ifp or not. will help when pim_ifp doesnt exist.

Reply:
Nah I don't think so :)  we need to gather this data no matter what.